src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -105,6 +124,22 @@ class GenerateHierarchy
         "ArborX::TreeConstruction::generate_hierarchy",
         Kokkos::RangePolicy<ExecutionSpace>(space, 0, leaf_nodes.extent(0)),
         *this);
+
+    Kokkos::deep_copy(
+        space,
+        Kokkos::View<BoundingVolume, Kokkos::HostSpace,
+                     Kokkos::MemoryUnmanaged>(&bounds),
+        Kokkos::View<BoundingVolume const, MemorySpace,
+                     Kokkos::MemoryUnmanaged>(getRootBoundingVolumePtr()));
+  }
+
+  KOKKOS_FUNCTION
+  BoundingVolume const *getRootBoundingVolumePtr() const
+  {
+    // Need address of the root node's bounding box to copy it back on the host,
+    // but can't access node elements from the constructor since the data is on
+    // the device.
+    return &_internal_nodes.data()->bounding_volume;
```

Comment:
Why did you prefer this approach as opposed to the one in the 3rd round of aprokop/ArborX#16
(`_bounds` member which is a mirror view of the `bounds` argument)?
Was the extra allocation causing slowdowns?  Is that why we are still using this hack to get the address of the bounding volume of the root node.
I am a bit surprised and if the performance is the same I would prefer the extra member.

Reply:
Using any extra variable inside the `operator()` was slowing things down for the smallest problem. We had 10K the smallest, but the relative slowdown would have been worse for smaller. Essentially, the thread taking the longest (i.e., traversing all the way to the root), was necessarily doing an extra Box assignment, which was non-trivial.

The only way to avoid that was to not use any extra variables. In which case, I ended up just deep copying.

Reply:
OK.  It might be worth expanding/updating the comment to explain that we chose to be naughty although we had alternatives for performance reasons.